File: POO/aula16/ProjetoAgencia/ContaPoupanca.cpp
#include "ContaPoupanca.h"

ContaPoupanca::ContaPoupanca(std::string cpf, std::string nome, int dia, int mes, int ano, float taxa):Conta(cpf, nome)
{
	this->setAniversario(Data(dia, mes, ano));
	this->setTaxa(taxa);
}

void ContaPoupanca::setAniversario(Data data)
{
	this->aniversario = data;
}

void ContaPoupanca::setTaxa(float taxa)
{
	this->taxa = taxa;
}

Data ContaPoupanca::getAniversario()
{
	return this->aniversario;
}

float ContaPoupanca::getTaxa()
{
	return this->taxa;
}

void ContaPoupanca::correcao()
{
	this->setSaldo(this->getSaldo() * 1.02);
}

void ContaPoupanca::extrato()
{
	Conta::extrato();
	std::cout << "ANIVERSARIO: " << this->getAniversario().getDia() << std::endl;
    std::cout << "===========================================" << std:: endl;
}

File: POO/aula16/JAgencia/ContaPoupanca.java
class ContaPoupanca extends Conta
{
	private Data aniversario;

	public ContaPoupanca()
	{	}

	public ContaPoupanca(String titular, String cpf, int dia, int mes, int ano)
	{
		super(titular, cpf);
		this.setAniversario(new Data(dia, mes, ano));
	}

	public void setAniversario(Data dt)
	{
		this.aniversario = dt;
	}

	public Data getAniversario()
	{
		return this.aniversario;
	}

	public void correcao()
	{
		this.setSaldo(this.getSaldo() * 1.02f);
	}

	public void extrato()
	{
		super.extrato();
		System.out.println("ANIVERSARIO : " + this.getAniversario().getDia());
		System.out.println("============================");
	}
}
